src/django_components/component.py

```diff
@@ -1234,6 +1238,11 @@ def _render_impl(
             )
         )
 
+        # The component rendering was short-circuited by an extension, skipping
+        # the rest of the rendering process. This may be for example a cached content.
+        if result_override is not None:
```

Comment:
One of the changes to make this work is that, if an extension returns a value from `on_component_input` hook, then we short-circuit the rendering, and return this  value instead.